crates/runtime/src/continuation.rs

```diff
@@ -393,7 +393,7 @@ pub fn suspend(instance: &mut Instance, tag_index: u32) {
     let running = unsafe {
         running
             .as_ref()
-            .expect("Calling suspend outside of a continuation")
+            .expect("Calling suspend outside of a continuation") // TODO(dhil): we should emit the trap UnhandledTag here.
```

Comment:
I think our confusion when debugging the sieve example shows that these should actually be two separate traps/errors: One for "There are one or more active handlers, but none of them handle the tag you suspended with" vs "you're calling suspend from the main stack".

Reply:
Good point

Reply:
Let's do the following then
```suggestion
            .expect("Calling suspend outside of a continuation") // TODO(dhil): we should trap here with a dedicated code indicating that we are suspending from the main stack
```

Reply:
I think it would suffice to just trap with `UnhandledTag` though. I am not opposed to trying something else.